r/tools/winlibs.R

```diff
@@ -28,19 +28,30 @@ if(!file.exists(sprintf("windows/arrow-%s/include/arrow/api.h", VERSION))){
     file.copy(localfile, "lib.zip")
   } else {
     # Download static arrow from rwinlib
-    if(getRversion() < "3.3.0") setInternet2()
-    try(download.file(sprintf("https://github.com/rwinlib/arrow/archive/v%s.zip", VERSION), "lib.zip", quiet = TRUE), silent = TRUE)
-    if(!file.exists("lib.zip")){
+    if (getRversion() < "3.3.0") setInternet2()
+    get_file <- function(template, version) {
+      try(download.file(sprintf(template, version), "lib.zip", quiet = TRUE), silent = TRUE)
+    }
+    # URL templates
+    # TODO: don't hard-code RTools 3.5? Can we detect which toolchain we have?
```

Comment:
```r
library(rvest)
read_html('https://dl.bintray.com/ursalabs/arrow-r/libarrow/bin')
  %>% html_nodes("a")
  %>% html_text()
  %>% as.character()
```
Yielded:
```
[1] "centos-7/"     "centos-8/"     "debian-10/"    "debian-9/"    
[5] "ubuntu-16.04/" "ubuntu-18.04/" "windows-35/"  
```
would it work to then grab `windows-${highestNumber}/`?

Reply:
No, it has to match the local toolchain version, and I don't know how to detect that within R, but it's fine, RTools 4.0 is still experimental. All released versions of R for the last few years are using 3.5. This is more a note for the future when 4.0 is released.